### ISSUE TITLE
False-positive for recur form inside an `if` and `core.match` form

### ISSUE THREAD

**Issue by keychera** [clj-kondo/clj-kondo#2661]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

clj-kondo --version                                   
clj-kondo v2025.09.22

**macro usage**

Related to clojure.core.match macro and loop-recur macro

**platform**

Native

**editor**

VSCode Calva version 2.0.540
(the clj-kondo bundled with calva in this version also experience the same bug)

**problem**

Unexpected lint-error of `Recur can only be used in tail position.` on valid recur position

**repro**

`file.clj`
```clojure
(require '[clojure.core.match :refer [match]])

(loop [[target & remaining] [{:hey "a"} {:hi "b"} {:hello "c"} {:hi "d"}]]
  (if (> (count remaining) 0)
    (match [target]
       [{:hey _}]
      (recur remaining) ;; <-- lint error here
      
      [{:hi _}]
      (do
        (println "hi" target)
        (recur remaining)) 

      :else (println "end"))
    (println "end")))
```

```
clj-kondo --lint .\file.clj
.\file.clj:7:7: error: Recur can only be used in tail position.
linting took 708ms, errors: 1, warnings: 0
```
note: if the `match` form is inside `when` or directly inside `loop`, no lint-error is reported.

**config**

No special config

**expected behavior**

no lint-error occurred
